core/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -86,6 +72,84 @@
         public static final String FETCH_SHARD_STORE = "fetch_shard_store";
     }
 
+    public enum ThreadPoolType {
+        CACHED("cached"),
+        DIRECT("direct"),
+        FIXED("fixed"),
+        SCALING("scaling");
+
+        private final String type;
+
+        public String getType() {
+            return type;
+        }
+
+        ThreadPoolType(String type) {
+            Objects.requireNonNull(type);
```

Comment:
Would be nice to have a better message with this assertion, perhaps "thread pool type must be one of [cached, direct, fixed, scaling]"